### ISSUE TITLE
Fix layer tests in `test-dist` (or skip them if `window`/`document` is required)

### ISSUE THREAD

**Issue by ericsoco** [uber/deck.gl#659]
There are multiple test scripts, but only `test-node` runs on pre-commit.

The recent EventManager/Hammer.js `window`/`document` server-side bug was not caught by `test-node` (which sets up JSDOM), but it _is_ caught by `test-dist` (which does no `window` or `document` mocking).

What exactly do we want to run and test on pre-commit? (We can take this conversation offline if need be but thought it would be good to record the answers here for posterity...)

**Comment by ericsoco** [comment]
It appears the tests introduced in https://github.com/uber/deck.gl/pull/611 pass when JSDOM is set up (via `npm run test-node`) but fail when JSDOM (and `window`/`document` mocks) are absent (via `npm run test-dist`).

I think that ^^ is the real issue to be addressed. Changing the issue name.

**Comment by shaojingli** [comment]
#660

**Comment by ibgreen** [comment]
> What exactly do we want to run and test on pre-commit? 

Apologies for the long answer, just to make sure everyone is on the same page:

First, the idea behind `test/node.js` and `test.node-dist.js`. 
* These are supposed to work identically
* The only difference being that one runs the test directly on the ES6 source and one runs the tests on the transpiled ES5 version of the library, just to make doubly sure the transpilation doesn't cause any issues.
* We should absolutely not import `jsdom` in just one of these two tests, they should both have the same setup, either neither or both should import it.

Second, about `jsdom` usage in node tests:
* The big problem is that the introduction of `jsdom` hides an important class of problems - i.e. we will no longer catch issues that happen under node in apps that don't include `jsdom` - which are very common (stray references to `window` and `document` etc). Thus, importing jsdom defeats an important part of the purpose of the node tests.
* I suspect the `jsdom` import into the node test script was introduced during the push to increase coverage, which also saw the addition of the `sinon` dependency.
* I'm not sure if the reason `jsdom` was included was because `sinon` requires it, or if it was due to another issue.
* Regardless, my strong preference would be that we find a way to avoid including `jsdom` in node tests. If it is indeed `sinon` that requires it, we might be able to replace `sinon` with our own spy function. I don't think we are using much of that library.
* If we absolutely need to include `jsdom` we might need to create a third node test script for node (`test/node-no-jsdom.js`?) that runs at least some tests without `jsdom` included.

Finally, in regards to the question about what tests do we want to run on test, pre-commit, travis CI, publish etc:

* For pre-commit, CI, publish, we want to run everything we got: the linter, run an error-free build, run multiple tests on node etc. Basically, we want to cast the widest possible net to prevent bugs from making their way into `master` and into the published packages. 

That said, for quick iteration of tests during development, it will be frustrating if too many tests are run each time. Since making sure that e.g. CI etc use all tests typically requires adding all the tests to the main `package.json` "test" script, in many modules we add a separate `test-fast` script that just runs one copy of the tests:
```
   "scripts": {
      "test"; "npm run lint && node test/node.js && node test/node-dist.js && .... "
      "test-fast" "node test/node.js",
      ...
   }
```

PS - We could possibly relax the pre-commit testing just a little if found too time consuming, but there is no reason to reduce the CI or prepublish testing.

**Comment by ericsoco** [comment]
Fixed by #660.
